src/lib.rs

```diff
@@ -580,6 +580,19 @@ impl<T> Receiver<T> {
         })
     }
 
+    /// Receives a message from the channel.
+    ///
+    /// Like [`Receiver::recv`], but uses a cloned `Receiver`.
+    ///
+    /// Good for using `Recv` in `Future` impls.
+    pub fn recv_owned(&self) -> RecvOwned<T> {
+        RecvOwned::_new(RecvOwnedInner {
+            receiver: self.clone(),
+            listener: None,
+            _pin: PhantomPinned,
+        })
+    }
+
```

Comment:
Add an example for this function, preferably one that points out where it could be useful.